File: GithubGists/app/src/main/java/ru/avystavkin/githubgists/screen/main/MainPageView.java
package ru.avystavkin.githubgists.screen.main;

import java.util.List;

import androidx.annotation.NonNull;
import ru.avystavkin.githubgists.models.local.Gist;
import ru.avystavkin.githubgists.models.local.User;
import ru.avystavkin.githubgists.screen.general.ErrorView;

public interface MainPageView extends ErrorView {

    void showGists(@NonNull List<Gist> gists);

    void showUsers(@NonNull List<User> users);
}

File: GithubGists/app/src/main/java/ru/avystavkin/githubgists/screen/main/MainPageView.kt
package ru.avystavkin.githubgists.screen.main

import ru.avystavkin.githubgists.models.database.Gist
import ru.avystavkin.githubgists.models.database.User
import ru.avystavkin.githubgists.screen.general.ErrorView

interface MainPageView : ErrorView {

    fun showGists(gists: List<Gist>)

    fun showUsers(users: List<User>)
}
